service/eventbridge/endpoints.go

```diff
@@ -624,3 +562,357 @@ func (r *resolver) ResolveEndpoint(
 	}
 	return endpoint, fmt.Errorf("endpoint rule error, %s", "Invalid Configuration: Missing Region")
 }
+
+type endpointParamsBinder interface {
+	bindEndpointParams(*EndpointParameters)
+}
+
+func bindEndpointParams(input endpointParamsBinder, options Options) *EndpointParameters {
+	params := &EndpointParameters{}
+
+	params.Region = aws.String(endpoints.MapFIPSRegion(options.Region))
+	params.UseDualStack = aws.Bool(options.EndpointOptions.UseDualStackEndpoint == aws.DualStackEndpointStateEnabled)
+	params.UseFIPS = aws.Bool(options.EndpointOptions.UseFIPSEndpoint == aws.FIPSEndpointStateEnabled)
+	params.Endpoint = options.BaseEndpoint
+
+	input.bindEndpointParams(params)
+
+	return params
+}
+
+func (in *ActivateEventSourceInput) bindEndpointParams(p *EndpointParameters) {
```

Comment:
Now that I see all of these empty bindings I'm thinking we probably want to just not generate in that case. The outer `bind` call could then delegate accordingly.

Reply:
Did the above after discussing offline.